content/courses/Ada_For_The_Embedded_C_Developer/chapters/02_Perspective.rst

```diff
@@ -1809,7 +1809,7 @@ a 32-bit modular type:
 
 [Ada]
 
-.. code:: ada run_button project=Courses.Ada_For_Embedded_C_Dev.Perspective.Overflow_Wraparound
+.. code:: ada run_button project=Courses.Ada_For_Embedded_C_Dev.Perspective.Overflow_Wraparound_2
```

Comment:
I agree with the distinction between the Ada and the C code. However, we should be able to use `Overflow_Wraparound_1` here as well (instead of giving it a new project name), as the current block will simply overwrite the previous one. Also, the `ada-run-expect-failure` class from the previous block shouldn't play a role here, as the class is block specific.

Reply:
When I change them I'll use the _Ada for both rather than creating two :+1: 

The reason for the need to have different C and Ada project names here change is because as it generates a `main.c` and `main.adb`, the two ada projects pick up this `main.c` and then gprbuild gives this error: `main.gpr:2:09: "main.adb" and "main.c" have the same object file name`